libtiledbsoma/src/soma/soma_group.h

```diff
@@ -0,0 +1,285 @@
+        std::optional<uint64_t> timestamp = std::nullopt);
+
+    /**
+     * @brief Open a group at the specified URI and return SOMAGroup
+     * object.
+     *
+     * @param mode TILEDB_READ or TILEDB_WRITE
+     * @param ctx TileDB context
+     * @param uri URI of the group
+     * @param name Name of the group
+     * @param timestamp Timestamp
+     * @return std::unique_ptr<SOMAGroup> SOMAGroup
+     */
+    static std::unique_ptr<SOMAGroup> open(
+        tiledb_query_type_t mode,
+        std::shared_ptr<Context> ctx,
+        std::string_view uri,
+        std::string_view name = "unnamed",
+        std::optional<uint64_t> timestamp = std::nullopt);
+
+    //===================================================================
+    //= public non-static
+    //===================================================================
+
+    /**
+     * @brief Construct a new SOMAGroup object.
+     *
+     * @param mode TILEDB_READ or TILEDB_WRITE
+     * @param uri URI of the group
+     * @param name Name of the group
+     * @param ctx TileDB context
+     * @param timestamp Timestamp
+     */
+    SOMAGroup(
+        tiledb_query_type_t mode,
+        std::string_view uri,
+        std::string_view name,
+        std::shared_ptr<Context> ctx,
+        std::optional<uint64_t> timestamp = std::nullopt);
+
+    SOMAGroup() = delete;
+    SOMAGroup(const SOMAGroup&) = delete;
+    SOMAGroup(SOMAGroup&&) = default;
+    ~SOMAGroup() = default;
+
+    /**
+     * Opens the SOMAGroup object.
+     *
+     * @param mode TILEDB_READ or TILEDB_WRITE
+     * @param timestamp Timestamp
+     */
+    void open(
+        tiledb_query_type_t mode,
+        std::optional<uint64_t> timestamp = std::nullopt);
+
+    /**
+     * Closes the SOMAGroup object.
+     */
+    void close();
+
+    /**
+     * Get the SOMAGroup URI.
+     */
+    std::string uri() const;
+
+    /**
+     * Get the Context associated with the SOMAGroup.
+     *
+     * @return std::shared_ptr<Context>
+     */
+    std::shared_ptr<Context> ctx();
+
+    /**
+     * Get a member from the SOMAGroup given the index.
+     *
+     * @param index of member
+     */
+    tiledb::Object get_member(uint64_t index) const;
+
+    /**
+     * Get a member from the SOMAGroup given the name.
+     *
+     * @param name of member
+     */
+    tiledb::Object get_member(const std::string& name) const;
+
+    /**
+     * Check if the SOMAGroup contains the given name.
+     *
+     * @param name of member
+     */
+    bool has_member(const std::string& name);
+
+    /**
+     * Add a named member to a SOMAGroup.
+     *
+     * @param uri of member to add
+     * @param relative is the URI relative to the SOMAGroup location
+     * @param name of member
+     */
+    void add_member(
+        const std::string& uri, bool relative, const std::string& name);
+
+    /**
+     * Get the number of members in the SOMAGroup.
+     */
+    uint64_t get_length() const;
```

Comment:
https://github.com/single-cell-data/SOMA/blob/main/abstract_specification.md#somacollection-1

I have this as length because the spec specifies a `get length` operation for `SOMACollection`, but I can change this to count if that name is better.

Reply:
(`SOMACollection` is going to be implemented by `SOMAGroup`.)

Reply:
Spec wins over Dirk, clearly.